### PR TITLE
Stop removing items that were included in the project file

### LINKED ISSUE #588: Content is not honored if it points to a file under Output
For the project below, the runtimeconfig.json is not packaged because it is a file under OutputPath, which gets excluded from Content.

My worry here is that once we take the new CLI, tools (ASP.NET) are going to be broken and are going to have to do “hacks” to get around this issue.
 
This also may seem somewhat arbitrary to users. They won’t be able to add any Content if that Content is coming from the output. I can see people complaining, not understanding and filing issues on us for this, as there is no real way to find out why this is happening.

```
<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0">
 
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp1.0</TargetFramework>
    <GeneratedPackageId>random-name</GeneratedPackageId>
    <PackageId>something</PackageId>
    <AssemblyName>dotnet-randompackage</AssemblyName>
  </PropertyGroup>
 
  <ItemGroup>
    <Content Include="$(OutputPath)\$(AssemblyName).runtimeconfig.json">
      <Pack>true</Pack>
      <PackagePath>lib\$(TargetFramework)</PackagePath>
    </Content>
  </ItemGroup>
 
  <ItemGroup>
    <PackageReference Include="Microsoft.NETCore.App" Version="1.0.1" />
  </ItemGroup>
 
</Project>
```

### LINKED ISSUE #623: Default items and default removes trample on item groups expressed in csproj
Can we make DefaultItems and DefaultRemoves smarter? **Current implementation tramples what the user expresses in their csproj file and requires deep understanding of internal workings of MSBuild to unwind this.**

Addition of Defaultitems and DefaultRemoves makes the behavior of ItemGroups in csproj unpredictable.

For example, consider this test application: 

```
Microsoft.DotNet.Tools.Watcher.FunctionalTests.csproj
Tests.cs
TestProjects/
     TestApp/
          TestApp.csproj
          Program.cs
```

In RC.2, my project defined these items:
```xml
<Compile Include="**\*.cs" Exclude="TestProjects\**\*" />
<Content Include="TestProjects\**\*" CopyToOutputDirectory="PreserveNewest" />
```

DefaultItems causes duplication in Compile groups, and default removes remove items I want in the Content group. To workaround this, I have to change the project to:

```xml
  <PropertyGroup>
    <DisableDefaultRemoves>true</DisableDefaultRemoves>
  </PropertyGroup>

  <ItemGroup>
    <Compile Remove="TestProjects\**\*" />

    <Compile Remove="$(DefaultRemoves)" />
    <EmbeddedResource Remove="$(DefaultRemoves)" />
    <None Remove="$(DefaultRemoves)" />
    <Content Remove="$(DefaultRemoves)" />

    <Content Include="TestProjects\**\*" CopyToOutputDirectory="PreserveNewest" />
 </ItemGroup>
```
This seems excessively ugly.

More details:
Here's the RC.2 project I'm having difficulty porting: https://github.com/aspnet/DotNetTools/blob/cb3c5d77ade283538af441be9792590422c789a5/test/Microsoft.DotNet.Watcher.Tools.FunctionalTests/Microsoft.DotNet.Watcher.Tools.FunctionalTests.csproj#L10

### PR DESCRIPTION
**Pull request by dsplaisted**
Instead, only apply the patterns that used to be called DefaultRemoves to the items that are implicitly included.  If someone adds an include for a file to their project file, we should respect that.

Fixes #623
Fixes #588

The `DefaultRemoves` property was originally used so that when you had wildcard includes in your project file, you wouldn't have to specifically exclude the `bin`, `obj`, etc. folders.  When we switched to implicit includes, I left it in with the idea that it would help you have a cleaner project file if you did want to explicitly see a wildcard include in your project, and it would help protect you from accidentally including items that you "shouldn't" from the output folders.

However, it's become apparent that it comes with two huge downsides:
 - If you actually want items that would match the patterns in `DefaultRemoves`, you will include them in the project file, and that won't work, and you will have absolutely no idea why and there's no good way to debug / figure it out
 - If you do manage to figure out why the items aren't being included, there's no good way to override the behavior to let you include a file.  You have to turn off the `DefaultRemoves` entirely, which then means your implicit globs will suddenly pick up a bunch of files that they shouldn't, unless you jump through even more hoops in your project file.

This PR makes it so that if you put an item in your project file, it will be respected.  The items that will be included implicitly remain the same as before.

This PR also takes the opportunity to remove the `packages` folder from the list of folders that the default includes don't apply to.  It's unclear why this was originally [added in the CLI](https://github.com/dotnet/cli/commit/94bafb17).  @davidfowl [had this to say about it](https://github.com/aspnet/websdk/pull/88/files/621bb3d2a625ceab3b2123b1a11dd69357eb79bd):

> It was likely because of a mix of older packages folder (nuget v2) and xproj. Though that should only have been at the solution level...

Removing this exclusion will make default items in projects with code or other assets in a "packages" folder work as people will expect.

This change will need to be made together with a change to the Web SDK, as now it will be its responsibility to remove these items from its default includes.  I will also submit a PR for that.

### CODE REVIEW

**Comment by dsplaisted** [comment]
I've filed the Web SDK PR: https://github.com/aspnet/websdk/pull/94

File: test/Microsoft.NET.Build.Tests/GivenThereAreDefaultItems.cs [hunk 1]
```diff
@@ -24,7 +24,7 @@ public void It_ignores_excluded_folders()
         {
             Action<GetValuesCommand> setup = getValuesCommand =>
             {
-                foreach (string folder in new[] { "bin", "obj", "packages" })
+                foreach (string folder in new[] { "bin", "obj" })
```
Comment (rainersigwald) [inline test/Microsoft.NET.Build.Tests/GivenThereAreDefaultItems.cs]: This seems fine but I'm a tiny bit worried about doing it in combination with the higher-pri change.

**Review by rainersigwald — APPROVED** [review]

**Review by nguerrera — APPROVED** [review]

**Review by srivatsn — APPROVED** [review]

File: test/Microsoft.NET.Build.Tests/GivenThereAreDefaultItems.cs (hunk 1 above)
Comment (dsplaisted) [inline test/Microsoft.NET.Build.Tests/GivenThereAreDefaultItems.cs]: Point taken, I'll back that part out and send it as a separate PR.

**Comment by dsplaisted** [comment]
@MattGertz for approval

#### Scenario

Include items in the project file that are in the output paths, or have an extension that we exclude from the implicit item includes

#### Bug

#623 and #588

#### Workarounds

See #623.  This is ugly and most people won't even get as far as figuring out what's wrong

#### Risk

Low - Tests should cover any impacted scenarios

#### Performance Impact

Low - the change moves a file filter from happening after the main contents of the project file is evaluated to before then

#### Regression Analysis

n/a

**Comment by MattGertz** [comment]
@dsplaisted @srivatsn I don't understand the full customer scenario here -- can you please add clarifications on the customer impact to the template?

**Comment by dsplaisted** [comment]
@MattGertz I've updated the scenario as requested

**Comment by natidea** [comment]
Shiproom approved to merge

**Pull request merged**